### ISSUE TITLE
Human-readable filename issue.

### ISSUE THREAD

**Issue by CiroGuangchengWen** [kelciour/awesometts-anki-addon#11]
![image](https://user-images.githubusercontent.com/54955984/70958788-208ec280-202f-11ea-85df-f2c6a49e8950.png)
The filename in cache. folder is still coded after the human-readable is checked.
Can you please help me out of it?
I want the audio file with the input text

Looking forward to your reply.
Thanks in advance.

**Comment by kelciour** [comment]
I just tried it and it worked for me. I guess, the issue can be fixed by deleting generated audio files from the AwesomeTTS cache folder. It can be done by clicking on the Advanced tab in the same AwesomeTTS Configuration window and clicking on "Delete Files" in the Caching group.

**Comment by NihilZero** [comment]
Thank you so much for your time.

I did what u mentioned. But still, the name of the audio file in the cache
folder is coded.
Is this because I generate this audio by using the preview function?
I'm using the Oddcast service by the way, can you please try it one more
time to see if it works for you?

Again, thanks alot.
Sincerely,

**Comment by kelciour** [comment]
I'm sorry but what do you mean by the cache folder? If it's about
collection.media folder, just use Anki - Tools - Check Media to delete
unused media files.

I tried it again from the card browser and it looks like there's no need to
delete audio files from the AwesomeTTS cache folder. The AwesomeTTS just
ignores hashed audio filename in the cache folder and generates a
human-readable audio filename, or maybe it's a bit more complex, but the
add-on works fine and the destination field contains a human-readable audio
filename.
I tried it again with the Preview button and it seems to work too.

And why Oddcast, why not Google Cloud Text-to-Speech? Just use the API key
from the add-on's page on AnkiWeb, if it's working, just don't overuse it.
The free tier is 1 million characters per month and it should be enough to
make the API key work every day without any problems for most of the users.
The only problem that I don't know what happens when the monthly free tier
is reached. Maybe the API key will be blocked forever or maybe just for the
rest of the month.

There's another alternative for AnkiMobile and AnkiDroid by using
on-the-fly TTS via JavaScript. I can't help with AnkiMobile -
https://anki.tenderapp.com/kb/card-appearance/text-to-speech-support - but
AnkiDroid is on my to-do list for sometime later. Though it might be worth
to check some TTS-related issues on the AnkiDroid GitHub page. But I don't
know how good they are.

ср, 18 дек. 2019 г. в 01:27, NihilZero <notifications@github.com>:

> Thank you so much for your time.
>
> I did what u mentioned. But still, the name of the audio file in the cache
> folder is coded.
> Is this because I generate this audio by using the preview function?
> I'm using the Oddcast service by the way, can you please try it one more
> time to see if it works for you?
>
> Again, thanks alot.
> Sincerely,
>

**Comment by kelciour** [comment]
Ah, sorry, I got what you're talking about. It's about `.cache` folder in `awesometts` subfolder. It indeed contains hashed audio filenames no matter what,, but `collection.media` folder will contain human-readable filenames.

I'm not very familiar with the add-on code, don't know if it's easy to fix or not, it's safe to fix and won't cause any other problems, but that's not something that I can do anytime soon. I've got a lot of work to do, some is already delayed by days, some by weeks, got a new one yesterday and there's another one scheduled for next week. That's not an issue as most of it just one-time work that can be done in half a day or a couple of days, but it'd take ~5 days to finish them all and I have other add-ons that needed to be updated sooner than later. Maybe it can be fixed in no time, just in 30 minutes, maybe not, but I'm not sure if it's worth to be fixed. I don't think I can spare any more time on it anytime soon. From my understanding, the .cache folder is not supposed to be used this way and there's nothing to fix.

As a workaround, I can only suggest, to export an Anki deck as apkg and import it in a new Anki profile, then open `collection.media` folder, or just open `collection.media` folder in the main Anki profile and sort files by date created. Something like this I guess should work similarly to .cache folder.
